File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int left = 0;
        int rollingSum = 0;
        int count = 0;

        for(int right = 0; right < arr.length; right++){
            rollingSum += arr[right];
            if(right-left+1 == k){
                if(rollingSum/k >= threshold) count++;
                rollingSum -= arr[left];
                left++;
            }
        }
        return count;
    }
}

File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        L = 0
        rolling_total = 0
        num_greater = 0

        for R in range(len(arr)):
            rolling_total += arr[R]
            if R-L+1 > k:
                rolling_total -= arr[L]
                L += 1
            if R-L+1 == k and rolling_total/k >= threshold: num_greater += 1
        
        return num_greater
